ext/afform/core/Civi/Api4/Action/Afform/AbstractProcessor.php

```diff
@@ -678,4 +680,22 @@ protected static function getNestedKey(array $values) {
     return is_array($firstValue) && $firstValue ? array_keys($firstValue)[0] : NULL;
   }
 
+  /**
+   * Replace relative date
+   * @param $records
+   */
+  protected function replaceRelativeDates($records) {
+    foreach ($records as $key => $record) {
+      foreach ($record['fields'] as $field => $value) {
+        if (str_contains($field, "date") && str_contains($value, ".")) {
```

Comment:
`str_contains($field, "date")` would also match "update", "validate", etc. Would it be better to use `_date`?

Reply:
Right, it's naive to assume all date fields have "date" in their name and non-date fields do not. We get into big trouble when we do stuff like that. It's similarly naive to assume a dot in the value always signifies a relative date.

Fortunately there's a better way. Take a look at the function `Civi\Api4\Utils\FormattingUtil::formatDateValue` which checks the `$value` for an exact match with the relative date filters like this: `array_key_exists($value, \CRM_Core_OptionGroup::values('relative_date_filters')` that would replace your `str_contains($value, ".")`.

For the `str_contains($field, "date")`, it's better to check the data-type of the field definition to see if it's a Date or Timestamp.